src/@chakra-ui/gatsby-plugin/components/components.utils.ts

```diff
@@ -70,3 +72,66 @@ export function defineMergeStyles(
 ): Record<string, SystemStyleObject> {
   return merge(defaultTheming, ...styleObjs)
 }
+
+export const _notDisabledReadOnly =
+  "&:not([data-disabled], [disabled], [data-readonly])"
+
+const $inputTriggerDisableColor = cssVar("input-trigger-disable-color")
+
+export const commonInputTriggerStyles = {
+  commonControlProps: {
+    border: "1px",
+    borderColor: "body.medium",
+    outline: "3px solid",
+    outlineColor: "transparent",
+    _checked: {
+      color: "background.base",
+      bg: "primary.base",
+      borderColor: "primary.base",
+    },
+    _focusVisible: {
+      borderColor: "primary.highContrast",
+      outlineColor: "primary.hover",
+      outlineOffset: "2px",
+      boxShadow: "none",
+    },
+    _disabled: {
+      bg: $inputTriggerDisableColor.reference,
+      borderColor: $inputTriggerDisableColor.reference,
+      opacity: 1,
+      _checked: {
+        bg: $inputTriggerDisableColor.reference,
+        borderColor: $inputTriggerDisableColor.reference,
+      },
+    },
+    [_notDisabledReadOnly]: {
+      // Hovering over the label triggers the style for the control
+      "*[data-checked]:hover > &": {
+        bg: "primary.hover",
+        borderColor: "primary.highContrast",
+      },
+      "*:not([data-checked]):hover > &": {
+        bg: "body.light",
+        borderColor: "primary.highContrast",
+      },
```

Comment:
@TylerAPfledderer I've updated the hover colors to match the DS. Not sure if there is a more concise way for these selectors but it works as expected.

We want to display `primary.hover` on checked and `body.light` on unchecked ones.